### PR TITLE
Support quantization and y flip options

### PR DESCRIPTION
**Pull request by rmarianski**
Connects to #33.

I'm not tied to the names for the options, they were just the first things that popped into my head. If you can think of better names, I'm happy to use those.

@loicgasser is this what you had in mind for the options and how it would work?

@zerebubuth could you review please too?

### CODE REVIEW

**Comment by zerebubuth** [comment]
From a technical point of view :+1:.

File: tests/test_encoder.py [hunk 1]
```diff
@@ -241,3 +241,37 @@ def test_encode_rounding_floats(self):
...
+        feature = dict(geometry=shape, properties=props)
+        features = [feature]
+        source = dict(name='layername', features=features)
+        bounds = 10.0, 10.0, 20.0, 20.0
+        pbf = encode(source, quantize_bounds=bounds)
+        result = decode(pbf)
+        act_feature = result['layername']['features'][0]
+        act_geom = act_feature['geometry']
+        exp_geom = [[2048, 2048]]
+        self.assertEqual(exp_geom, act_geom)
```
Comment (loicgasser) [inline tests/test_encoder.py]: what about a combinaison of `quantize_bounds` and `y_coord_down` ?

And use a point that is not excatly in the middle of the bounds to test the impact of `y_coord_down` parameter

**Comment by loicgasser** [comment]
It's awesome you guys provide this option, thanks a lot. (this is what I had in mind)

I just have one small remark concering the tests.

In general what is the convention the `snake_like` coding with `_` or the `camelCase` or as you please kind of convention.

**Comment by loicgasser** [comment]
Also be careful doc is messed up see interpreted doc:

https://github.com/mapzen/mapbox-vector-tile/blob/support-quantization-and-yflip-options/README.rst

**Comment by loicgasser** [comment]
Also https://github.com/mapzen/mapbox-vector-tile/blob/master/mapbox_vector_tile/__init__.py#L5

I think this should be optional.

File: tests/test_encoder.py (hunk 1 above)
Comment (rmarianski) [inline tests/test_encoder.py]: Added a test for both that isn't in the middle in 7c1df44.

**Comment by rmarianski** [comment]
> In general what is the convention the snake_like coding with _ or the camelCase or as you please kind of convention.

I'm trying to stick to [pep8](https://www.python.org/dev/peps/pep-0008/). Unfortunately the unittest module doesn't adhere to that, so it can look a little inconsistent when using it.

> Also be careful doc is messed up see interpreted doc:

Nice catch! I just transitioned it to markdown since that's more common in my experience. The formatting should look ok now.

> I think this should be optional.

I added support for passing in custom extents in 5832c0d.

**Comment by nvkelso** [comment]
Is this ready to merge?

**Comment by rmarianski** [comment]
I'll go ahead and merge it. If we want to make further changes, let's open a new issue for it.

**Pull request merged**